pkg/controller/federatedcluster/controller.go

```diff
@@ -78,15 +76,19 @@ type ClusterController struct {
 }
 
 // StartClusterController starts a new cluster controller.
-func StartClusterController(config *util.ControllerConfig, stopChan <-chan struct{}, clusterMonitorPeriod time.Duration) {
+func StartClusterController(config *util.ControllerConfig, stopChan <-chan struct{}, clusterMonitorPeriod time.Duration) error {
 	fedClient, kubeClient, crClient := config.AllClients("cluster-controller")
-	controller := newClusterController(fedClient, kubeClient, crClient, config.FederationNamespaces, clusterMonitorPeriod)
+	controller, err := newClusterController(config, fedClient, kubeClient, crClient, clusterMonitorPeriod)
```

Comment:
If config is passed in, is there a reason not to invoke `AllClients` in `newClusterController`?